src/merge.rs

```diff
@@ -192,3 +194,295 @@ impl<'repo> Drop for AnnotatedCommit<'repo> {
+    their_label: Option<CString>,
+
+    // raw data
+    raw: raw::git_merge_file_options,
+}
+
+impl Default for MergeFileOptions {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl MergeFileOptions {
+    /// Creates a default set of merge options.
+    pub fn new() -> MergeFileOptions {
+        let mut opts = MergeFileOptions {
+            ancestor_label: None,
+            our_label: None,
+            their_label: None,
+            raw: unsafe { mem::zeroed() },
+        };
+        assert_eq!(
+            unsafe { raw::git_merge_file_options_init(&mut opts.raw, 1) },
+            0
+        );
+        opts
+    }
+
+    /// Specify ancestor label, default is "ancestor"
+    pub fn ancestor_label<T: IntoCString>(&mut self, t: T) -> &mut MergeFileOptions {
+        self.ancestor_label = Some(t.into_c_string().unwrap());
+
+        self.raw.ancestor_label = self
+            .ancestor_label
+            .as_ref()
+            .map(|s| s.as_ptr())
+            .unwrap_or(ptr::null());
+
+        self
+    }
+
+    /// Specify ancestor label, default is "ours"
+    pub fn our_label<T: IntoCString>(&mut self, t: T) -> &mut MergeFileOptions {
+        self.our_label = Some(t.into_c_string().unwrap());
+
+        self.raw.our_label = self
+            .our_label
+            .as_ref()
+            .map(|s| s.as_ptr())
+            .unwrap_or(ptr::null());
+
+        self
+    }
+
+    /// Specify ancestor label, default is "theirs"
+    pub fn their_label<T: IntoCString>(&mut self, t: T) -> &mut MergeFileOptions {
+        self.their_label = Some(t.into_c_string().unwrap());
+
+        self.raw.their_label = self
+            .their_label
+            .as_ref()
+            .map(|s| s.as_ptr())
+            .unwrap_or(ptr::null());
+
+        self
+    }
+
+    /// Specify a side to favor for resolving conflicts
+    pub fn file_favor(&mut self, favor: FileFavor) -> &mut MergeFileOptions {
+        self.raw.favor = favor.convert();
+        self
+    }
+
+    /// Specify marker size, default is 7: <<<<<<< ours
+    pub fn marker_size(&mut self, size: u16) -> &mut MergeFileOptions {
+        self.raw.marker_size = size;
+        self
+    }
+
+    /// Acquire a pointer to the underlying raw options.
+    pub unsafe fn raw(&self) -> *const raw::git_merge_file_options {
+        &self.raw as *const _
+    }
+}
+
+/// For git_merge_file_input
+pub struct MergeFileInput {
+    raw: raw::git_merge_file_input,
+
+    /// File name of the conflicted file, or `NULL` to not merge the path.
+    ///
+    /// You can turn this value into a `std::ffi::CString` with
+    /// `CString::new(&entry.path[..]).unwrap()`. To turn a reference into a
+    /// `&std::path::Path`, see the `bytes2path()` function in the private,
+    /// internal `util` module in this crate’s source code.
+    path: Option<CString>,
+
+    /// File mode of the conflicted file, or `0` to not merge the mode.
+    pub mode: Option<FileMode>,
```

Comment:
These fields should be private so alterations are forced to go through public functions.

Reply:
Also I think it's ok to eschew storing `mode` here, we can just store it directly in the `raw` field.

Reply:
Yes, fixed that